bin/diagnostics.bash

```diff
@@ -308,15 +325,102 @@ get_config_files() {
     done
 }
 
+run_diagnostics_default_metrics() {
+    token=$(fetch_oidc_token)
+    domain="https://kube.$(yq4 '.global.opsDomain' "${config["config_file_sc"]}"):6443"
+    endpoint="${domain}/api/v1/namespaces/thanos/services/thanos-query-query-frontend:9090/proxy/api/v1"
+    header="Authorization: Bearer ${token}"
+    range_arg="--data-urlencode start=$(date -d -"${1}" -Iseconds) --data-urlencode end=$(date -Iseconds) --data-urlencode step=1m"
+
+    query_and_parse() {
+        query="${1}"
+        print_func="${2}"
+        # shellcheck disable=SC2086
+        res=$(curl "${endpoint}/query_range" -k -s --header "${header}" --data-urlencode query="${query}" $range_arg)
+        if [[ $(echo "${res}" | jq 'select(.data.result != [])') ]]; then
```

Comment:
First avoid `echo "${variable} | command`, use `command <<< "${variable}"`,
second be explicit about what you check here,
third wrap `$(command)` in quotes everywhere except for simple `for var in $(list-generator); do`where you can be _absolutely sure that the data will never contain spaces_, for all other times use `readarray` first.

Reply:
What is the issue with `echo "${variable} | command`? (seeing as its everywhere in our bin already)

Reply:
It's not really an issue but a general recommendation. It is more optimised, as it doesn't create a process, and it makes debugging easier when tracing enabled.
And the fact that it is everywhere is because we essentially never refactor anything.

Reply:
> second be explicit about what you check here
What do you mean exactly? What I want to check is if the array is empty or not. How do I do that more explicitly here?

Reply:
> > second be explicit about what you check here
> 
> What do you mean exactly? What I want to check is if the array is empty or not. How do I do that more explicitly here?

You have a test `[[ ]]` where you only provide the output of a command, which means it defaults to a `-n` check.
If that is what you want to do add the `-n` to make it explicit.

Reply:
Ah okay I see what you mean now. I'll try to make it more explicit in that regard.

Reply:
Think I have adressed all of your comments now.